File: 1929_concatenation_of_array/concatenation_of_arrays.js
// JS: https://leetcode.com/problems/concatenation-of-array/submissions/1177760773/
// TS: https://leetcode.com/problems/concatenation-of-array/submissions/1177760357/
function concat_array(nums) {
    let end = nums.length * 2;
    let index = nums.length - 1;
    for(let i = 0; nums.length < end; i++) {
        index++;
        nums[index] = nums[i];
    }

    return nums;
}

console.log(concat_array([1,2,3]));

File: 1929_concatenation_of_array/concatenation_of_arrays.py
# Py3: https://leetcode.com/problems/concatenation-of-array/submissions/1177768293/
from typing import List
def concat_arrays(nums: List[int]) -> List[int]:
    for i in range(len(nums)):
        nums.append(nums[i])
    
    return nums

print(concat_arrays([1,2,3]))